### ISSUE TITLE
All engines time out resulting in no search results (docker DNS issue)

### ISSUE THREAD

**Issue by voronind-com** [searx/searx#3405]
**Version of Searx, commit number if you are using on master branch and stipulate if you forked Searx**
Latest broken version: 1.1.0-32-806bd804 
Latest working version: 1.0.0-349-a1645851

**How did you install Searx?**
official docker images from Docker Hub

**What happened?**
Search not working at all.

**How To Reproduce**
1. Try to search for anything. 
2. See no search suggestions (when enabled).
3. Press enter to search.
4. Wait about 2 sec for default timeout.
5. See all engine timeout errors with no results.

**Expected behavior**
Basic usage working.

**Screenshots & Logs**
![image](https://user-images.githubusercontent.com/22127600/200089627-1830b0b1-3c59-449c-a16a-d6c4c56ef4f7.png)

logs (after reproducing the issue):
```
search  | searx version 1.1.0-32-806bd804
search  | 
search  | Use existing /etc/searx/uwsgi.ini
search  | Use existing /etc/searx/settings.yml
search  | Listen on 0.0.0.0:8080
search  | [uWSGI] getting INI configuration from /etc/searx/uwsgi.ini
search  | [uwsgi-static] added mapping for /static => /usr/local/searx/searx/static

```

docker-compose file:
```
version: '3.8'
services:
  search: 
#    image: docker.io/searx/searx:1.0.0-349-a1645851
    image: docker.io/searx/searx:latest
    container_name: search
    restart: unless-stopped
    user: "0:0"
    cpus: 1.0
    mem_limit: 1G
    logging:
#      driver: "none"
      options:
        max-file: "1"
        max-size: "1m"
    networks:
      docker: 
        ipv4_address: 10.1.0.26
    volumes:
      - ./data:/etc/searx
    environment:
      - BASE_URL=[REMOVED]

networks:
  docker: 
    name: docker
    external: true
```

**Additional context**
I'm using basic configs, aka clean docker install.
Just switching to old working version does the job.
I update quarterly and experiencing this issue for half a year now (3 mo ago I've decided to skip one update).

**Comment by voronind-com** [comment]
duplicate of #3042

**Comment by voronind-com** [comment]
I've figured it out for Docker.
I found scrapper logs inside the container (why they are not printed to docker logs?). and the issue was with name resolution. the fix was not to use docker's default DNS server and specify it explicitly. i.e. in docker-compose file you write:
```
dns:
  - 1.1.1.1
```
or (my specific case, this is dns address specified system-wide in resolv.conf)
```
dns:
  - 10.1.0.6
```

after this it all works. second variant is my host's dns provider, which is used by docker as system-wide, but somehow only works when you specify it like that and not automatically. hope this helps someone. the reasons of why this happens are unknown to me. I can cooperate to help finding the core issue.

edit: maybe searx somehow blocks docker's default dnf subnet? it's 127.0.0.11.

edit: after this fix, I still get brave search timeout tho. don't know if this is related.
